lib/trino-filesystem/src/main/java/io/trino/filesystem/fileio/BulkOperationsFileIo.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.fileio;
+
+import io.trino.filesystem.TrinoFileSystem;
+import org.apache.iceberg.io.BulkDeletionFailureException;
+import org.apache.iceberg.io.SupportsBulkOperations;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulkOperationsFileIo
+        extends ForwardingFileIo
```

Comment:
I would implement this directly in ForwardingFileIo, not as a subclass.

(that would probably also satisfy https://github.com/trinodb/trino/pull/15981/files#r1097092879, right?)

cc @electrum